packages/server/src/effection/restartable.ts

```diff
@@ -2,10 +2,10 @@ import { spawn, Context, Operation } from 'effection'
 import { subscribe, ChainableSubscription } from '@effection/subscription';
 import { Slice } from "@bigtest/atom";
 
-export function *restartable<T>(slice: Slice<T, unknown>, operation: (t: T) => Operation<void>): Operation<void> {
+export function *restartable<S, A>(slice: Slice<S, A>, operation: (t: S) => Operation<void>): Operation<void> {
   let current: Context | null = null;
 
-  let subscription: ChainableSubscription<T, undefined> = yield subscribe(slice);
+  let subscription: ChainableSubscription<S, undefined> = yield subscribe(slice);
 
```

Comment:
I'm ok refactoring the types to be more expressive, but is this necessary for this change?

Reply:
@cowboyd agreed and reverted.